monai/losses/unified_focal_loss.py

```
@@ -162,79 +241,72 @@ def __init__(
-        if torch.max(y_true) != self.num_classes - 1:
-            raise ValueError(f"Please make sure the number of classes is {self.num_classes-1}")
-
-        n_pred_ch = y_pred.shape[1]
-        if self.to_onehot_y:
-            if n_pred_ch == 1:
-                warnings.warn("single channel prediction, `to_onehot_y=True` ignored.")
-            else:
-                y_true = one_hot(y_true, num_classes=n_pred_ch)
+            is_binary_logits = y_pred.shape[1] == 1 and not self.use_softmax
+            if not self.to_onehot_y and not is_binary_logits:
+                raise ValueError(f"ground truth has different shape ({y_true.shape}) from input ({y_pred.shape})")
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)